### PR TITLE
[MNT] Adding Scorecard.yml Workflow and Scorecard Badge GOSST

### PR DESCRIPTION
**Pull request by PatriceJada**
#### What does this implement/fix? Explain your changes.
This implementation integrates the OpenOSSF Scorecard, which provides an overview of the security health of the repository. The Scorecard analyzes the repository based on various security metrics, including code review, dependencies, CI/CD configurations, and more.

#### Any other comments?
Part of [GSOC ](https://github.com/numfocus/gsoc/issues/470)

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Comment by MatthewMiddlehurst** [comment]
Thanks for the PR and best of luck on your GSoC project. Let us know if you need any help from our side.

**Comment by MatthewMiddlehurst** [comment]
By the look of things requires https://github.com/google/oss-fuzz/pull/12053 first?

**Comment by PatriceJada** [comment]
@MatthewMiddlehurst No, this is its own separate thing

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
Comment (MatthewMiddlehurst) [inline .github/workflows/scorecard.yml:27]: This does not really follow the structure we used in other workflows to action versioning, I think it's fine to just have `actions/checkout@v4`. Same for other action versions.

File: .github/workflows/scorecard.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
+# by a third-party and are governed by separate terms of service, privacy
+# policy, and support documentation.
+
+name: Scorecard supply-chain security
+on:
+  branch_protection_rule:
+  schedule:
+    - cron: '42 16 * * 5'
```
Comment (MatthewMiddlehurst) [inline .github/workflows/scorecard.yml:9]: We usually run these early in the morning UTC i.e. `'30 1 * * 7'`. I imagine it is fast and wont make much of a different though.

**Review by MatthewMiddlehurst — COMMENTED** [review]
Looks fine, but have a few comments. The badge currently shows nothing, but I assume that will change when a report is uploaded?

**Review by MatthewMiddlehurst — APPROVED** [review]
LGTM. I think the pre-commit action does not like the force pushes, but not really relevant 🙂.

**Pull request merged**